engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkStatusDetailsDeterminer.scala

```diff
@@ -23,36 +24,39 @@ class FlinkStatusDetailsDeterminer(
   def statusDetailsFromJobOverviews(jobOverviews: List[JobOverview]): Future[Map[ProcessName, List[StatusDetails]]] =
     Future
       .sequence {
-        jobOverviews
-          .groupBy(_.name)
-          .flatMap { case (name, jobs) =>
-            namingStrategy.decodeName(name).map(decoded => (ProcessName(decoded), jobs))
+        for {
+          job  <- jobOverviews
+          name <- namingStrategy.decodeName(job.name).map(ProcessName(_))
+        } yield withParsedJobConfig(job.jid, name).map { jobConfigOpt =>
+          val details = jobConfigOpt.map { jobConfig =>
+            StatusDetails(
+              SimpleStateStatus.fromDeploymentStatus(toDeploymentStatus(JobStatus.valueOf(job.state), job.tasks)),
+              jobConfig.deploymentId,
+              Some(ExternalDeploymentId(job.jid)),
+              version = Some(jobConfig.version),
+              startTime = Some(job.`start-time`),
+              attributes = Option.empty,
+              errors = List.empty
+            )
+          } getOrElse {
+            logger.debug(
+              s"No correct job config in deployed scenario: $name. Returning ${SimpleStateStatus.DuringDeploy} without version"
+            )
+            StatusDetails(
+              SimpleStateStatus.DuringDeploy,
+              // For scheduling mechanism this fallback is probably wrong // TODO: switch scheduling mechanism deployment ids to UUIDs
+              Some(DeploymentId(job.jid)),
+              Some(ExternalDeploymentId(job.jid)),
+              version = None,
+              startTime = Some(job.`start-time`),
+              attributes = Option.empty,
+              errors = List.empty
+            )
           }
-          .map { case (name, jobs) =>
-            val statusDetails = jobs.map { job =>
-              withParsedJobConfig(job.jid, name).map { jobConfig =>
-                // TODO: return error when there's no correct version in process
```

Comment:
I changed this behaviour because this solution with returning `None` in `version` for this case causes flickering of `ProblemStateStatus.missingDeployedVersion`.  It happens less often for remote flink, but for mini cluster more often as it faster start tasks.